platforms/ide/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/model/IsolatedGradleProjectInternal.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugins.ide.internal.tooling.model;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.tooling.internal.gradle.DefaultProjectIdentifier;
+import org.gradle.tooling.internal.gradle.GradleProjectIdentity;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Represents a Gradle project, isolated from the project hierarchy.
+ * <p>
+ * <b>This model is internal, and is NOT part of the public Tooling API.</b>
+ */
+@NonNullApi
+public class IsolatedGradleProjectInternal implements Serializable, GradleProjectIdentity {
```

Comment:
💭 Does this type need to implement `GradleProjectIdentity`?

Reply:
Good catch!